sandbox/grist/dropdown_condition.py

```diff
@@ -43,35 +39,33 @@ def perform_dropdown_condition_renames(useractions, renames):
       continue
 
     # Find out what table this column refers to and belongs to.
-    if col.type.startswith("Ref:") :
-      ref_table_id = col.type[4:] 
-    elif col.type.startswith("RefList:"):
-      ref_table_id = col.type[8:] 
-    else:
-      ref_table_id = None
+    ref_table_id = usertypes.get_referenced_table_id(col.type)
     self_table_id = col.parentId.tableId
 
     def renamer(subject):
       table_id = ref_table_id if subject.type == "choiceAttr" else self_table_id
-      # Dropdown conditions stay in widgetOptions, even when the current column type can't make use of them.
-      # Attributes of "choice" do not make sense for columns that are not Ref and RefList.
-      # We set ref_table_id to None in this case, so table_id will be None for stray choiceAttrs, therefore
-      # the subject will not be renamed.
+      # Dropdown conditions stay in widgetOptions, even when the current column type can't make
+      # use of them. Thus, attributes of "choice" do not make sense for columns other than Ref and
+      # RefList, but they may exist.
+      # We set ref_table_id to None in this case, so table_id will be None for stray choiceAttrs,
+      # therefore the subject will not be renamed.
       # Columns of "rec" are still renamed accordingly.
       return renames.get((table_id, subject.name))
-      
+
     new_dc_formula = predicate_formula.process_renames(dc_formula, _DCEntityCollector(), renamer)
 
-    widget_options["dropdownCondition"] = {"text": new_dc_formula}
-    try:
-      # Parse the new dropdown condition formula if it is syntactically correct.
+    # The data engine stops processing remaining formulas when it hits an exception. Parsing a
```

Comment:
This sentence sounds wrong. Generally Grist formulas can have exceptions and that doesn't normally interfere with other formulas. So probably you mean something more specific -- it would be good to clarify.

Reply:
I rephrased it a bit. Hope it's clearer now.